mne/io/pick.py

```diff
@@ -1,1199 +1,1202 @@
-        elif pick in _FNIRS_CH_TYPES_SPLIT:
-            fnirs |= {pick}
-        else:
-            bad_type = pick
-            break
-    else:
-        # triage MEG and FNIRS, which are complicated due to non-bool entries
-        extra_picks = set()
-        if len(meg) > 0 and not kwargs.get('meg', False):
-            # easiest just to iterate
-            for use_meg in meg:
-                extra_picks |= set(pick_types(
-                    info, meg=use_meg, ref_meg=False, exclude=exclude))
-        if len(fnirs) > 0 and not kwargs.get('fnirs', False):
-            if len(fnirs) == 1:
-                kwargs['fnirs'] = list(fnirs)[0]
-            else:
-                kwargs['fnirs'] = list(fnirs)
-        picks_type = pick_types(info, exclude=exclude, **kwargs)
-        if len(extra_picks) > 0:
-            picks_type = sorted(set(picks_type) | set(extra_picks))
-
-    #
-    # finally: ensure we have exactly one usable list
-    #
-    all_picks = (picks_generic, picks_name, picks_type)
-    any_found = [len(p) > 0 for p in all_picks]
-    if sum(any_found) == 0:
-        if not allow_empty:
-            raise ValueError(
-                'picks (%s) could not be interpreted as '
-                'channel names (no channel "%s"), channel types (no '
-                'type "%s"), or a generic type (just "all" or "data")'
-                % (repr(orig_picks) + extra_repr, bad_name, bad_type))
-        picks = np.array([], int)
-    elif sum(any_found) > 1:
-        raise RuntimeError('Some channel names are ambiguously equivalent to '
-                           'channel types, cannot use string-based '
-                           'picks for these')
-    else:
-        picks = np.array(all_picks[np.where(any_found)[0][0]])
-    if return_kind:
-        picked_ch_type_or_generic = not len(picks_name)
-        return picks, picked_ch_type_or_generic
-    return picks
-
-
-def _pick_inst(inst, picks, exclude, copy=True):
-    """Return an instance with picked and excluded channels."""
-    if copy is True:
-        inst = inst.copy()
-    picks = _picks_to_idx(inst.info, picks, exclude=[])
-    pick_names = [inst.info['ch_names'][pick] for pick in picks]
-    inst.pick_channels(pick_names)
-
-    if exclude == 'bads':
-        exclude = [ch for ch in inst.info['bads']
-                   if ch in inst.info['ch_names']]
-    if exclude is not None:
-        inst.drop_channels(exclude)
-    return inst
-
-
-def _get_channel_types(info, picks=None, unique=False, only_data_chs=False):
-    """Get the data channel types in an info instance."""
-    none = 'data' if only_data_chs else 'all'
-    picks = _picks_to_idx(info, picks, none, (), allow_empty=False)
-    ch_types = [channel_type(info, pick) for pick in picks]
-    if only_data_chs:
-        ch_types = [ch_type for ch_type in ch_types
-                    if ch_type in _DATA_CH_TYPES_SPLIT]
-    return set(ch_types) if unique is True else ch_types
+# -*- coding: utf-8 -*-

```

Comment:
@evgenygoldstein we have a problem here. Your editor just reformatted this file and the diff is now unreadable.